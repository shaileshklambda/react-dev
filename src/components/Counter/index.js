// @flow
import React from "react";
import styles from "./index.module.css";
import { Row, Col } from "react-bootstrap";

const Counter = props => (
  <div>
    <Row>
      <Col md={4}>
        <div className={styles.rectangle}>
          <Row>
            <Col md={12}>
              <span className={styles.title}>
                {props.counter && props.counter.title
                  ? props.counter.title
                  : "N/A"}
              </span>
            </Col>
          </Row>
          <Row>
            <Col md={2}>
              <span className={styles.count}>
                {props.counter && props.counter.count ? props.counter.count : 0}
              </span>
            </Col>
          </Row>
        </div>
      </Col>
    </Row>
  </div>
);

export default Counter;
