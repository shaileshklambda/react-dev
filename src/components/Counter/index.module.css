.Counter {
  display: block;
}

.rectangle {
  border-radius: 6px;
  border: 1px solid #e2e2e2;
  padding: 10px 22px;
  background-color: #fff;
}

.title {
  width: 115px;
  height: 17px;
  font-family: Lato;
  font-size: 14px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
}
.count {
  width: 36px;
  height: 36px;
  font-family: Lato;
  font-size: 30px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.4px;
  color: #0ebac5;
}
