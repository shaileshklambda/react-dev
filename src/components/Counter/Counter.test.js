import React from "react";
import renderer from "react-test-renderer";
import Counter from "./index.js";

it("Counter: default", () => {
  const component = renderer.create(<Counter />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
