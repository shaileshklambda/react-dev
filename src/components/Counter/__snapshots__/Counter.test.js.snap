// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter: default 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div
        className="rectangle"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <span
              className="title"
            >
              N/A
            </span>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-2"
          >
            <span
              className="count"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
